splice_machine/src/main/java/com/splicemachine/derby/impl/load/HdfsImport.java

```diff
@@ -670,18 +715,31 @@ private static void doImport(String schemaName,
                 throw new SQLException(e);
             }
 
+            String insertMode = (isUpsert ? "UPSERT" : "INSERT");
+            if( isLoadReplaceMode ) insertMode = InsertNode.LOAD_REPLACE;
+
             ColumnInfo columnInfo = new ColumnInfo(conn, schemaName, tableName, insertColumnList);
             String selectList = generateColumnList(lcc,schemaName,tableName,insertColumnList, true, null);
             String vtiTable = importVTI + " AS importVTI (" + columnInfo.getImportAsColumns() + ")";
             String insertSql = "INSERT INTO " + entityName + "(" + columnInfo.getInsertColumnNames() + ") " +
-                    "--splice-properties useSpark=true , insertMode=" + (isUpsert ? "UPSERT" : "INSERT") + ", statusDirectory=" +
+                    "--splice-properties useSpark=true , insertMode=" + insertMode + ", statusDirectory=" +
                     badRecordDirectory + ", badRecordsAllowed=" + badRecordsAllowed + ", bulkImportDirectory=" + bulkImportDirectory
                     + ", samplingOnly=" + samplingOnly + ", outputKeysOnly=" + outputKeysOnly + ", skipSampling=" + skipSampling
                     + (skipConflictDetection ? ", skipConflictDetection=true" : "") + (skipWAL ? ", skipWAL=true" : "")
                     + (indexName !=null ? (", index=" + indexName):"") + "\n" +
                     " SELECT "+ selectList +
                     " from " + vtiTable;
 
+            if(isLoadReplaceMode)
+            {
+                // delete table before inserting
+                try (PreparedStatement ips = conn.prepareStatement("DELETE FROM " + entityName + DeleteNode.NO_TRIGGER_RI_PROPERTY)) {
```

Comment:
Are we deleting all rows? Should we use truncate here instead?

Reply:
My understanding is that not all rows are to be deleted from the table. Should join the table with input file using PK as join key, and delete all rows from the table whose PK is in the join result.

Reply:
discussed with Gene, he thinks the DELETE+INSERT is the thing the customer wants. We can add an UPSERT mode later if needed.
TRUNCATE would be possible, but is more difficult to implement. As it is a performance improvement over this, i created DB-10642 .